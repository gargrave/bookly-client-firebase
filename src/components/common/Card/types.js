// @flow
export type CardTextListItem = {
  title?: string,
  value?: string,
}
