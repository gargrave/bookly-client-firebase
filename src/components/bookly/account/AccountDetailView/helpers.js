// @flow
import { format } from 'date-fns';

import type { User } from '../../../../globals/flowtypes';

export const userBasicDetailsTextList = (user: User) => {
  const {
    email,
    emailVerified,
  } = user;

  return [
    { title: 'Email', value: email },
    { title: 'Verified', value: `${emailVerified}` },
  ];
};

export const userRegDetailsTextList = (user: User) => {
  const {
    lastLogin,
    registered,
  } = user;

  return [
    { title: 'Registered', value: format(registered, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Last login', value: format(lastLogin, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};
