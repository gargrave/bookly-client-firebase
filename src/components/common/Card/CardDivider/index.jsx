// @flow
import React from 'react';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
};

const CardDivider = ({}: Props) => {
  return (
    <hr className={buildClasses(['card__divider'])} />
  );
};

CardDivider.propTypes = {
};

export default CardDivider;
