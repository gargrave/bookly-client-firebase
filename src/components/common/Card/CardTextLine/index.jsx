// @flow
import React from 'react';
import { bool, string } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
  bold?: boolean,
  condensed?: boolean,
  text: string,
};

const CardTextLine = ({
  bold = false,
  condensed = false,
  text,
}: Props) => {
  return (
    <p
      className={buildClasses([
        'card__text',
        bold ? 'card__text--bold' : '',
        condensed ? 'condensed' : '',
      ])}
    >
      {text}
    </p>
  );
};

CardTextLine.propTypes = {
  bold: bool,
  condensed: bool,
  text: string.isRequired,
};

export default CardTextLine;
