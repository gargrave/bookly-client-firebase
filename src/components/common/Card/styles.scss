@import '../../../styles/_index';

.bookly-card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 600px;
  padding: 10px 20px;

  &--top-margin-med {
    margin-top: 50px;
  }

  hr {
    margin: 12px 0;
  }

  &--hoverable:hover {
    background: #e4effc;
    cursor: pointer;
  }

  &__header,
  &__title,
  &__text {
    margin: 0;
  }

  &__header,
  &__title {
    font-weight: bold;
  }

  &__header {
    font-size: 1.5rem;
  }

  &__text {
    color: #6e6969;
  }

  &__text + &__text {
    margin-top: 8px;
  }
}

.bookly-detail-card {
  padding: 15px;

  .bookly-card__title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .bookly-card__title + .bookly-card__text {
    font-size: 1.1rem;
  }
}

.bookly {
  &-list-detail-card {
    padding: 10px
  }
}
