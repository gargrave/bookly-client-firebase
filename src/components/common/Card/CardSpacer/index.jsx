// @flow
import React from 'react';
import { oneOf } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

import './styles.css';

type Props = {
  size?: string,
};

const CardSpacer = ({
  size = 'medium',
}: Props) => {
  return (
    <div className={buildClasses(`card__spacer--${size}`)} />
  );
};

CardSpacer.propTypes = {
  size: oneOf(['small', 'medium', 'large']),
};

export default CardSpacer;
