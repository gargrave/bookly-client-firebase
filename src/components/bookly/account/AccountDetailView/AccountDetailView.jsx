// @flow
import React from 'react';
import { bool, func, instanceOf, oneOfType, shape, string } from 'prop-types';

import type { User } from '../../../../constants/flowtypes';

import { buildClasses } from '../../../../utils/cssHelpers';

import {
  userBasicDetailsTextList,
  userRegDetailsTextList,
} from './helpers';

import Button from '../../../common/Button';
import ButtonRow from '../../../common/ButtonRow';
import Card from '../../../common/Card';
import CardSpacer from '../../../common/Card/CardSpacer';
import CardTextList from '../../../common/Card/CardTextList';

type Props = {
  onLogoutClick: Function,
  user: User,
};

const AccountDetailView = ({
  onLogoutClick,
  user,
}: Props) => {
  return (
    <div className={buildClasses(['detail-view', 'account-detail-view'])}>
      <Card
        classes={['detail-card', 'account-detail-card']}
        header="My Account"
        hoverable={false}
      >

        <CardSpacer />
        <CardTextList textList={userBasicDetailsTextList(user)} />
        <CardSpacer />
        <CardTextList textList={userRegDetailsTextList(user)} />
        <CardSpacer />

        <ButtonRow>
          <Button
            onClick={onLogoutClick}
            position="left"
            text="Logout"
            type="info"
          />
        </ButtonRow>
      </Card>
    </div>
  );
};

AccountDetailView.propTypes = {
  onLogoutClick: func.isRequired,
  user: shape({
    email: string,
    emailVerified: bool,
    lastLogin: oneOfType([
      instanceOf(Date),
      string,
    ]),
    registered: oneOfType([
      instanceOf(Date),
      string,
    ]),
  }).isRequired,
};

export default AccountDetailView;
