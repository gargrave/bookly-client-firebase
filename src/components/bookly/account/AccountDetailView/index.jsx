// @flow
import React from 'react';
import { bool, func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { User } from '../../../../constants/flowtypes';

import { buildClasses } from '../../../../utils/cssHelpers';

import Button from '../../../common/Button';
import ButtonRow from '../../../common/ButtonRow';
import Card from '../../../common/Card';
import CardSpacer from '../../../common/Card/CardSpacer';
import CardTextList from '../../../common/Card/CardTextList';

type Props = {
  onLogoutClick: Function,
  user: User,
};

const userBasicDetailsTextList = (user: User) => {
  const {
    email,
    emailVerified,
  } = user;

  return [
    { title: 'Email', value: email },
    { title: 'Verified', value: `${emailVerified}` },
  ];
};

const userRegDetailsTextList = (user: User) => {
  const {
    lastLogin,
    registered,
  } = user;

  return [
    { title: 'Registered', value: format(registered, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Last login', value: format(lastLogin, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};

const AccountDetailView = ({
  onLogoutClick,
  user,
}: Props) => {
  return (
    <div className={buildClasses(['detail-view', 'account-detail-view'])}>
      <Card
        classes={['detail-card', 'account-detail-card']}
        header="My Account"
        hoverable={false}
      >

        <CardSpacer />
        <CardTextList textList={userBasicDetailsTextList(user)} />
        <CardSpacer />
        <CardTextList textList={userRegDetailsTextList(user)} />
        <CardSpacer />

        <ButtonRow>
          <Button
            onClick={onLogoutClick}
            position="left"
            text="Logout"
            type="info"
          />
        </ButtonRow>
      </Card>
    </div>
  );
};

AccountDetailView.propTypes = {
  onLogoutClick: func.isRequired,
  user: shape({
    email: string,
    emailVerified: bool,
    lastLogin: oneOfType([
      instanceOf(Date),
      string,
    ]),
    registered: oneOfType([
      instanceOf(Date),
      string,
    ]),
  }).isRequired,
};

export default AccountDetailView;
