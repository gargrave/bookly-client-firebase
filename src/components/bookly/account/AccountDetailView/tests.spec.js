import React from 'react';
import { shallow } from 'enzyme';

import { userMocks } from '../../../../utils/mocks/';

import Card from '../../../common/Card';

import AccountDetailView from './';

const defaultProps = Object.freeze({
  onLogoutClick: jest.fn(),
  user: Object.create(userMocks[0]),
});

function getComponent(extraProps = {}) {
  const props = Object.assign({}, defaultProps, extraProps);
  return shallow(<AccountDetailView {...props} />);
}

describe('AccountDetailView', () => {
  let component;

  test('matches the snapshot', () => {
    component = getComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = getComponent();
    expect(component.find('.bookly-account-detail-view')).toHaveLength(1);
    expect(component.find(Card)).toHaveLength(1);
  });
});
