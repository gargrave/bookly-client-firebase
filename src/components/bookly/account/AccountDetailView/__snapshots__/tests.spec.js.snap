// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountDetailView
    onLogoutClick={[Function]}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        classes={
          Array [
            "detail-card",
            "account-detail-card",
          ]
        }
        hoverable={false}
        title="My Account"
      >
        <CardTextList
          textList={
            Array [
              Object {
                "title": "Email",
                "value": "mockuser1@gmail.com",
              },
            ]
          }
        />
        <CardDivider />
        <CardTextList
          textList={
            Array [
              Object {
                "title": "Registered",
                "value": "Jun. 30, 2017, 12:52:54",
              },
              Object {
                "title": "Last login",
                "value": "Jun. 30, 2017, 18:11:03",
              },
            ]
          }
        />
        <CardDivider />
        <ButtonRow>
          <Button
            onClick={[Function]}
            position="left"
            text="Logout"
            type="info"
          />
        </ButtonRow>
      </Card>,
      "className": "bookly-detail-view bookly-account-detail-view",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Email",
                  "value": "mockuser1@gmail.com",
                },
              ]
            }
          />,
          <CardDivider />,
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Registered",
                  "value": "Jun. 30, 2017, 12:52:54",
                },
                Object {
                  "title": "Last login",
                  "value": "Jun. 30, 2017, 18:11:03",
                },
              ]
            }
          />,
          <CardDivider />,
          <ButtonRow>
            <Button
              onClick={[Function]}
              position="left"
              text="Logout"
              type="info"
            />
          </ButtonRow>,
        ],
        "classes": Array [
          "detail-card",
          "account-detail-card",
        ],
        "hoverable": false,
        "title": "My Account",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "textList": Array [
              Object {
                "title": "Email",
                "value": "mockuser1@gmail.com",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "textList": Array [
              Object {
                "title": "Registered",
                "value": "Jun. 30, 2017, 12:52:54",
              },
              Object {
                "title": "Last login",
                "value": "Jun. 30, 2017, 18:11:03",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              onClick={[Function]}
              position="left"
              text="Logout"
              type="info"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "position": "left",
              "text": "Logout",
              "type": "info",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          classes={
            Array [
              "detail-card",
              "account-detail-card",
            ]
          }
          hoverable={false}
          title="My Account"
        >
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Email",
                  "value": "mockuser1@gmail.com",
                },
              ]
            }
          />
          <CardDivider />
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Registered",
                  "value": "Jun. 30, 2017, 12:52:54",
                },
                Object {
                  "title": "Last login",
                  "value": "Jun. 30, 2017, 18:11:03",
                },
              ]
            }
          />
          <CardDivider />
          <ButtonRow>
            <Button
              onClick={[Function]}
              position="left"
              text="Logout"
              type="info"
            />
          </ButtonRow>
        </Card>,
        "className": "bookly-detail-view bookly-account-detail-view",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Email",
                    "value": "mockuser1@gmail.com",
                  },
                ]
              }
            />,
            <CardDivider />,
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Registered",
                    "value": "Jun. 30, 2017, 12:52:54",
                  },
                  Object {
                    "title": "Last login",
                    "value": "Jun. 30, 2017, 18:11:03",
                  },
                ]
              }
            />,
            <CardDivider />,
            <ButtonRow>
              <Button
                onClick={[Function]}
                position="left"
                text="Logout"
                type="info"
              />
            </ButtonRow>,
          ],
          "classes": Array [
            "detail-card",
            "account-detail-card",
          ],
          "hoverable": false,
          "title": "My Account",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "textList": Array [
                Object {
                  "title": "Email",
                  "value": "mockuser1@gmail.com",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "textList": Array [
                Object {
                  "title": "Registered",
                  "value": "Jun. 30, 2017, 12:52:54",
                },
                Object {
                  "title": "Last login",
                  "value": "Jun. 30, 2017, 18:11:03",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                onClick={[Function]}
                position="left"
                text="Logout"
                type="info"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "position": "left",
                "text": "Logout",
                "type": "info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
